tomb-common/src/banyan_api/client.rs

```diff
@@ -165,41 +165,52 @@ impl Client {
         }
     }
 
-    /// Call a method that implements ApiRequest
+    /// Call a method that implements ApiRequest on the core server
     pub async fn call<T: ApiRequest>(
         &mut self,
         request: T,
     ) -> Result<T::ResponseType, ClientError> {
+        // Determine if this request requires authentication
         let add_authentication = request.requires_authentication();
+        // Create the request builder
         let mut request_builder = request.build_request(&self.remote_core, &self.reqwest_client);
+        // If we need authentication
         if add_authentication {
+            // Obtain the bearer token
```

Comment:
We generally don't need these detailed comments, the code itself is describing what is going on. Please reserve comments for when there is an assumption, edge case being made or considered that may not be immediately obvious from the code, around code that might seem useless or redundant at first glance that may be accidentally cleaned up in haste later on, or when something more long winded is needed like a discussion on trade offs.